src/plugins/googledrive/server/services/operation-service.js

```diff
@@ -0,0 +1,148 @@
+      }
+      return res.files;
+    } else {
+      const driveResponse = await drive.files
+        .get({
+          fileId: ctx.request.query.fileId,
+          fields: '*',
+          supportsAllDrives: true,
+        })
+        .then((driveResponseList) => driveResponseList.data)
+        .catch((err) => err);
+      return {
+        id: driveResponse.id,
+        name: driveResponse.name,
+        trashed: driveResponse.trashed,
+        modifiedTime: driveResponse.modifiedTime,
+        webViewLink: driveResponse.webViewLink,
+      };
+    }
+  },
+  upload: async (ctx) => {
+    if (!ctx.request.files) {
+      return ctx.badRequest('No files were uploaded');
+    }
+
+    const SERVICE_ACCOUNT_PATH = path.join(
+      process.cwd(),
+      'config',
+      'env',
+      'google_drive_service_account.json'
+    );
+
+    const SCOPES = ['https://www.googleapis.com/auth/drive.file'];
+
+    const auth = new google.auth.GoogleAuth({
+      keyFile: SERVICE_ACCOUNT_PATH,
+      scopes: SCOPES,
+    });
+    const drive = google.drive({ version: 'v3', auth });
+
+    const {
+      request: { body, files: { files } = {} },
+    } = ctx;
+    const uploadSingleFile = async (fileName, fileType, filePath) => {
+      const fsnp = require('fs');
+
+      const fileMetadata = {
+        name: fileName,
+        parents: [process.env.FOLDERID], //WORKING
+        mimeType: fileType, //mime-types to get the file types
+      };
+      const media = {
+        mimeType: fileType,
+        body: fsnp.createReadStream(filePath),
+      };
+      try {
+        const createResponse = await drive.files.create({
+          requestBody: fileMetadata,
+          media: media,
+          supportsAllDrives: true,
+          enforceSingleParent: true,
+          fields: 'id, name,parents,webViewLink,mimeType',
+        });
+
+        if (!createResponse.data.parents.includes(process.env.FOLDERID)) {
+          throw new Error('File was not created in the expected folder.');
+        }
+        return createResponse.data;
+      } catch (error) {
+        return error;
+      }
+    };
+
+    try {
+      const uploadResponse = await uploadSingleFile(
+        files['name'],
+        files['type'],
+        files.path
+      );
+      return uploadResponse;
+    } catch (err) {
+      ctx.throw(533, err);
+      return err;
+    }
+  },
+  deleteFile: async (ctx) => {
+    const SERVICE_ACCOUNT_PATH = path.join(
+      process.cwd(),
+      'config',
+      'env',
+      'google_drive_service_account.json'
+    );
+
+    const SCOPES = ['https://www.googleapis.com/auth/drive.file'];
+
+    const auth = new google.auth.GoogleAuth({
+      keyFile: SERVICE_ACCOUNT_PATH,
+      scopes: SCOPES,
+    });
+
+    const drive = google.drive({ version: 'v3', auth });
+    // To send the file to trash
+    const body_value = {
+      trashed: true,
+    };
+    try {
+      const response = await drive.files.update({
+        fileId: ctx.request.query.fileId,
```

Comment:
If we want to address this comment https://github.com/dev-launchers/strapiv4/pull/250#discussion_r1881345303, we will need to get the file ID from `ctx.request.params` https://docs-v4.strapi.io/dev-docs/backend-customization/requests-responses#ctxrequest.